src/java.naming/share/classes/com/sun/jndi/ldap/sasl/LdapSasl.java

```diff
@@ -123,15 +125,23 @@ public static LdapResult saslBind(LdapClient clnt, Connection conn,
         try {
             // Prepare TLS Channel Binding data
             if (conn.isTlsConnection()) {
-                TlsChannelBindingType cbType =
-                        TlsChannelBinding.parseType(
+                TlsChannelBindingType cbType;
+                try {
+                    cbType = TlsChannelBinding.parseType(
                                 (String)env.get(TlsChannelBinding.CHANNEL_BINDING_TYPE));
+                } catch (ChannelBindingException e) {
+                    throw new SaslException(e.getMessage());
```

Comment:
How about setting `e` as cause of new exception? In `TlsChannelBinding.java` the when the original exception was thrown (the 2nd throws) there was a cause.

Reply:
Agreed.